src/arrayMethodJoin.js

```diff
@@ -5,7 +5,24 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    if (arguments.length !== 0 && separator !== undefined) {
+      let resultStr = '';
+
+      for (let i = 0; i < this.length; i++) {
+        const item = (this[i] === undefined || this[i] === null)
+          ? '' : this[i];
```

Comment:
```suggestion
        if (this[i] !== undefined && this[i] !== null) {
            resultStr += this[i];
        }
```
you don't need 'item' variable, you can check 'if this[i] !== undefined && this[i] !== null', if true, then resultStr += this[i]